dotnet/src/SemanticKernel/Reliability/DefaultHttpRetryHandler.cs

```diff
@@ -0,0 +1,256 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Configuration;
+
+namespace Microsoft.SemanticKernel.Reliability;
+
+internal sealed class DefaultHttpRetryHandler : DelegatingHandler
```

Comment:
I defaulted to `internal`, but I'm thinking now `public` might be best so folks who use `OpenAITextCompletion` or others directly could also leverage this class. Curious what your thoughts are @dluc

Reply:
I'm changing to public for now. Still interested in thoughts/opinions.